application/ledger_query_service/src/ethereum/queries.rs

```diff
@@ -408,63 +493,200 @@ mod tests {
+                data: None,
+                topics: vec![],
+            }],
         };
-        let log1 = log(vec![redeem_log_msg]);
+
+        let log1 = log(
+            CONTRACT_ADDRESS.into(),
+            vec![REDEEM_LOG_MSG.into()],
+            Bytes(vec![]),
+        );
         let receipt1 = transaction_receipt(vec![log1]);
         assert_that!(query1.matches_transaction_receipt(receipt1)).is_false();
 
-        let query2 = LogQuery { topics: vec![] };
-        let log2 = log(vec![redeem_log_msg]);
+        let query2 = EventQuery {
+            event_matchers: vec![],
+        };
+
+        let log2 = log(
+            CONTRACT_ADDRESS.into(),
+            vec![REDEEM_LOG_MSG.into()],
+            Bytes(vec![]),
+        );
         let receipt2 = transaction_receipt(vec![log2]);
         assert_that!(query2.matches_transaction_receipt(receipt2)).is_false()
     }
 
     #[test]
     fn given_a_transaction_receipt_should_match_two_log_query() {
-        let redeem_log_msg =
-            "0xB8CAC300E37F03AD332E581DEA21B2F0B84EAAADC184A295FEF71E81F44A7413".into();
-        let random_log_msg =
-            "0xB8CAC300E37F03AD332E581DEA21B2F0B84EAAADC184A295FEF71E81F44A7412".into();
-
-        let query = LogQuery {
-            topics: vec![vec![redeem_log_msg], vec![random_log_msg]],
+        let query = EventQuery {
+            event_matchers: vec![
+                EventMatcher {
+                    address: Some(CONTRACT_ADDRESS.into()),
+                    data: None,
+                    topics: vec![Some(REDEEM_LOG_MSG.into())],
+                },
+                EventMatcher {
+                    address: Some(CONTRACT_ADDRESS.into()),
+                    data: None,
+                    topics: vec![Some(RANDOM_LOG_MSG.into())],
+                },
+            ],
         };
 
-        let log1 = log(vec![redeem_log_msg]);
-        let log2 = log(vec![random_log_msg]);
+        let log1 = log(
+            CONTRACT_ADDRESS.into(),
+            vec![REDEEM_LOG_MSG.into()],
+            Bytes(vec![]),
+        );
+        let log2 = log(
+            CONTRACT_ADDRESS.into(),
+            vec![RANDOM_LOG_MSG.into()],
+            Bytes(vec![]),
+        );
 
         let receipt = transaction_receipt(vec![log1, log2]);
 
         assert_that!(query.matches_transaction_receipt(receipt)).is_true()
     }
 
     #[test]
-    fn given_a_transaction_receipt_should_not_match_very_two_topic_query() {
-        let redeem_log_msg =
-            "0xB8CAC300E37F03AD332E581DEA21B2F0B84EAAADC184A295FEF71E81F44A7413".into();
-        let random_log_msg =
-            "0xB8CAC300E37F03AD332E581DEA21B2F0B84EAAADC184A295FEF71E81F44A7412".into();
-
-        let query = LogQuery {
-            topics: vec![vec![redeem_log_msg, random_log_msg]],
+    fn given_a_transaction_receipt_with_address_should_not_match_with_different_address() {
+        let query = EventQuery {
+            event_matchers: vec![EventMatcher {
```

Comment:
:wrench: It might be worth creating a little builder API for the `EventMatcher` to make the tests less verbose. Something like:

```rust
fn for_contract<S: Into<String>>(address: S) -> Self {
...
}

fn with_topic<S: Into<String>>(self, topic: S) -> Self {
...
}
```

Reply:
It doesn't have to handle all the cases (like `None` topics) just the most common ones :)